confluent/docker_utils/cub.py

```diff
@@ -80,14 +80,17 @@ def __request(host, port, secure, ignore_cert, path = ""):
         secure: Use TLS to secure the connection.
         ignore_cert: Ignore TLS certificate errors.
         path: Path on the remote server.
+        username: Username used to authenticate to the server.
+        password: Password used to authenticate to the server.
 
     Returns:
         Request result.
 
     """
     scheme = "https" if secure else "http"
     url = "%s://%s:%s/%s" % (scheme, host, port, path)
-    return requests.get(url, verify = not ignore_cert)
+    auth = (username, password) if (username is not None or password is not None) else None
```

Comment:
```suggestion
    auth = (username, password) if (username is not None and password is not None) else None
```

I think this needs to be an `and` not `or` - otherwise you can pass combinations of `username=None password=foo` / `username=foo password=None` - I can't seem to find anything specific.

Secondary to this, this is just BasicAuth right? If so lets do what the example does here:

https://docs.python-requests.org/en/master/user/authentication/#basic-authentication

and wrap it in `HTTPBasicAuth('user', 'pass')` so its clear that these options trigger BasicAuth flows.